### PR TITLE
Not recommend setting tiflash logical split since 6.2

### PR DESCRIPTION
**Pull request by JaySon-Huang**
Signed-off-by: JaySon-Huang <tshent@qq.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

According to https://github.com/pingcap/tiflash/issues/5576, if user deploys tiflash with explicit setting `profiles.default.dt_enable_logical_split = 1`.
After several logical split, PageStorage V3 will throw an exception for the third time logical split and make TiFlash crash.

Here we declare the config item to be deprecated

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.2 (TiDB 6.2 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/10840
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by flowbehappy — APPROVED** [review]

File: tiflash/tiflash-configuration.md [hunk 1]
```diff
@@ -173,11 +173,12 @@ delta_index_cache_size = 0
...
 
 [profiles.default]
-    ## The default value is false. This parameter determines whether the segment
+    ## [Deprecated] The default value is false. This parameter determines whether the segment
     ## of DeltaTree Storage Engine uses logical split.
     ## Using the logical split can reduce the write amplification.
     ## However, these are at the cost of disk space waste.
-    ## Modifying the default value is not recommended.
+    ## Note that logical split has compatibility issues with other features since v6.2.0 and
+    ## it is not recommended to set the value to true.
```
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:181]:
```suggestion
    ## Note that logical split has known issue on v6.2.0 and later versions,
    ## it is strongly not recommended to set the value to true.
```

**Comment by JaySon-Huang** [comment]
PTAL again @shichun-0415 @flowbehappy

File: tiflash/tiflash-configuration.md [hunk 2]
```diff
@@ -177,7 +177,9 @@ delta_index_cache_size = 0
     ## of DeltaTree Storage Engine uses logical split.
     ## Using the logical split can reduce the write amplification.
     ## However, these are at the cost of disk space waste.
-    ## Modifying the default value is not recommended.
+    ## Note that logical split has known issue on v6.2.0 and later versions
+    ## [#5576](https://github.com/pingcap/tiflash/issues/5576),
+    ## it is strongly not recommended to set the value to `true`.
```
Comment (shichun-0415) [inline tiflash/tiflash-configuration.md:182]:
```suggestion
    ## It is strongly not recommended to set `dt_enable_logical_split` to `true` in v6.2.0
    ## and later versions. For details, see known issue [#5576](https://github.com/pingcap/tiflash/issues/5576).
```

**Review by shichun-0415 — APPROVED** [review]

**Pull request merged**